### PR TITLE
[SYCL] Set -O0 as the default SYCL device optimization if -g is passed

### PR DESCRIPTION
**Pull request by srividya-sundaram**
Set `O0` as the default SYCL device optimization if `-g` is passed and optimization level is not provided explicitly.
Use of `'-g'` without any optimization-level option will turn off compiler optimizations similar to use of `'-O0'`.

### CODE REVIEW

**Review by sarnex — APPROVED** [review]
esimd lgtm

**Review by mdtoguchi — APPROVED** [review]
LGTM - Thanks.

File: clang/test/Driver/sycl-device-optimizations.cpp [hunk 1]
```diff
@@ -53,3 +53,59 @@
...
+// RUN:   %clang -### -fsycl -O2 -g %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHECK-G-O2 %s
+// For clang_cl, -O2 maps to -O3
+// RUN:   %clang_cl -### -fsycl -O2 -g %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHECK-G-O3 %s
+// CHECK-G-O2: clang{{.*}} "-fsycl-is-device{{.*}}" "-O2"
+// CHECK-G-O2: sycl-post-link{{.*}} "-O2"
+// CHECK-G-O2-NOT: "-O0"
+// CHECK-G-O3: clang{{.*}} "-fsycl-is-device{{.*}}" "-O3"
+// CHECK-G-O3: sycl-post-link{{.*}} "-O3"
```
Comment (mdtoguchi) [inline clang/test/Driver/sycl-device-optimizations.cpp:76]: Side note - the `-O2` to `-O3` behavior is not in xmain, so some adjustments to the test will be needed when this is pulled down.

**Comment by srividya-sundaram** [comment]
@intel/llvm-gatekeepers Hello, This is ready for merging. Thanks!

**Pull request merged**